cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java

```diff
@@ -0,0 +1,116 @@
+import com.google.common.collect.ImmutableSet;
+import com.google.common.io.Files;
+import org.apache.twill.filesystem.Location;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.io.IOException;
+
+/**
+ * Unit tests with authorization enabled.
+ */
+public class AuthorizationTest extends TestBase {
+  private static Location extensionJar;
+  static {
+    LocalLocationFactory locationFactory = new LocalLocationFactory(Files.createTempDir());
+    try {
+      extensionJar = AppJarHelper.createDeploymentJar(locationFactory, InMemoryAuthorizer.class);
+    } catch (IOException e) {
+      throw Throwables.propagate(e);
+    }
+  }
+  private InstanceId instance;
+
+  private final Principal alice = new Principal("alice", Principal.PrincipalType.USER);
+
+  @ClassRule
+  public static final TestConfiguration TEST_CONFIG = new TestConfiguration(
+    Constants.Security.ENABLED, true,
+    Constants.Security.Authorization.ENABLED, true,
+    Constants.Security.Authorization.EXTENSION_JAR_PATH, extensionJar.toURI().getPath()
+  );
+
+  @BeforeClass
+  public static void setup() {
+    SecurityRequestContext.setUserId("alice");
+  }
+
+  @Before
+  public void setupTest() throws Exception {
+    instance = new InstanceId(getConfiguration().get(Constants.Router.ADDRESS));
+    getAuthorizationManager().grant(NamespaceId.DEFAULT, alice, ImmutableSet.of(Action.ADMIN));
+  }
+
+  @Test
+  public void testNamespaces() throws Exception {
+    NamespaceId namespace = new NamespaceId("authorization");
+    try {
+      createNamespace(namespace.toId());
+      Assert.fail("Namespace create should have failed because alice is not authorized");
+    } catch (UnauthorizedException expected) {
+      // expected
+    }
+    getAuthorizationManager().grant(instance, alice, ImmutableSet.of(Action.ADMIN));
+    createNamespace(namespace.toId());
+    // No authorization currently for listing and retrieving namespace
+    listNamespaces();
+    getNamespace(namespace);
+    // revoke admin privilege
+    getAuthorizationManager().revoke(namespace, alice, ImmutableSet.of(Action.ADMIN));
+    try {
+      deleteNamespaceDatasets(namespace);
+      Assert.fail("Namespace delete datasets should have failed because alice's admin privileges have been revoked");
+    } catch (UnauthorizedException expected) {
+      // expected
+    }
+    // grant privileges again
+    getAuthorizationManager().grant(namespace, alice, ImmutableSet.of(Action.ADMIN));
+    deleteNamespaceDatasets(namespace);
+    // deleting datasets does not revoke privileges.
+    updateNamespaceProperties(new NamespaceMeta.Builder(NamespaceMeta.DEFAULT).setDescription("new desc").build());
+    deleteNamespace(namespace.toId());
+  }
+
+  @AfterClass
+  public static void cleanup() throws Exception {
+    // we want to execute TestBase's @AfterClass before unsetting userid
```

Comment:
Unsetting the userId seemed like a good thing to do, so we return to the same state we were in at the beginning of the test. It should be done after TestBase's ``@AfterClass``, so the cleanup can happen while alice is still ADMIN - in ``@BeforeClass``, we give alice ADMIN privileges to the ``default`` namespace. 

However, to make it even better, we should reset the user id to the one that existed before this test, instead of resetting it to null. Will fix.